client/src/pages/thread/thread.jsx

```diff
@@ -49,11 +74,18 @@ const Thread = () => {
   );
 
   const handleToggleShowOwnPosts = useCallback(() => {
-    postsFilter.userId = showOwnPosts ? userId : undefined;
-    postsFilter.from = 0;
-    handlePostsLoad(postsFilter);
-    postsFilter.from = postsFilter.count; // for the next scroll
-  }, [userId, showOwnPosts, handlePostsLoad]);
+    const currentUserId = showOwnPosts ? userId : undefined;
+
+    handlePostsLoad({ ...postsFilterInitialState, userId: currentUserId });
+
+    dispatchPostsFilter({
+      type: PostsFilterAction.TOGGLE_SHOW_OWN_POSTS,
+      payload: {
+        userId: currentUserId,
+        from: postsFilter.count
+      }
+    });
```

Comment:
On my opinion we can have only one dispatch there and trigger load posts via useEffect

Reply:
And let's move from and count into global store